modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectStateFlowUtil.java

```diff
@@ -97,18 +97,18 @@ private static ObjectStateTransition _toObjectStateTransition(
 			ListTypeEntryLocalServiceUtil.fetchListTypeEntry(
 				listTypeDefinitionId, objectStateTransition.getKey());
 
-		if ((objectStateFlowId == 0) && (sourceObjectStateId == 0)) {
+		ObjectState targetObjectState =
```

Comment:
I can see how this improves the code on the scenario where the objectStateFlowId would be 0, but i think this change allows the scenario where the sourceObjectStateId is 0 to still go through. 

That specific scenario happens when we are receiving an object through batch, and so we have the Stateflows but not the Ids set.

Could you try using batch to create an object that has a stateFlow attached and see if the states and the transitions are created as intended?